### PR TITLE
OpenVX media extension - add support for the HIP GPU backend

### PR DESCRIPTION
**Pull request by asalmanp**

### CODE REVIEW

File: amd_openvx_extensions/amd_media/decoder.cpp [hunk 1]
```diff
@@ -762,14 +840,55 @@ void CLoomIoMediaDecoder::DecodeLoop(int mediaIndex)
...
-                    cl_int err = clEnqueueWriteBuffer(cmdq, mem[bufId], CL_TRUE, clOffset + mediaIndex * decoderImageHeight * clStride, decoderImageHeight * clStride, tmp_frame->data, 0, nullptr, nullptr);
+#if ENABLE_OPENCL
+                    cl_int err = clEnqueueWriteBuffer(cmdq, (cl_mem)mem[bufId], CL_TRUE, gpuOffset + mediaIndex * decoderImageHeight * gpuStride, decoderImageHeight * gpuStride, tmp_frame->data, 0, nullptr, nullptr);
                     if (err < 0) {
                         vxAddLogEntry((vx_reference)node, VX_FAILURE, "ERROR: clEnqueueWriteBuffer(buf[%d], slice[%d]) failed (%d)\n", bufId, mediaIndex, err);
                         continue;
                     }
                     clFinish(cmdq);
+#elif ENABLE_HIP
+                    if (!hip_dev_prop.canMapHostMemory) {
```
Comment (rrawther) [inline amd_openvx_extensions/amd_media/decoder.cpp:884]: Looks like Line#866 is already copying the data to device. Do we need to do it again here?

Comment (asalmanp) [reply amd_openvx_extensions/amd_media/decoder.cpp:884]: yes, these are for different conditions. line#866 is for this condition if (conversionContext[mediaIndex] != NULL) {
and line#885 is for the else condition..

**Comment by kiritigowda** [comment]
@asalmanp there is a merge conflict

**Comment by asalmanp** [comment]
> @asalmanp there is a merge conflict

@kiritigowda it's fixed now.

**Pull request merged**